lib/ace/ext/papeeria/katex-previewer.coffee

```diff
@@ -398,53 +398,52 @@ define((require, exports, module) ->
+      popoverHandler.destroy()
+      editor.off("changeSelection", sh.hideSelectionPopover)
+      editor.getSession().off("changeScrollTop", sh.hideSelectionPopover)
+      editor.getSession().off("changeScrollLeft", sh.hideSelectionPopover)
+
+    renderSelectionUnderCursor: ->
+      cursorPos = editor.getCursorPosition()
+      cursorPosition = editor.renderer.textToScreenCoordinates(cursorPos.row, cursorPos.column)
+      popoverPosition = {
+        top: "#{cursorPosition.pageY + 24}px"
+        left: "#{cursorPosition.pageX}px"
+      }
+      content = katex.renderToString(
+        editor.getSelectedText(),
+        KATEX_OPTIONS
+      )
+      popoverHandler.show("Preview", content, popoverPosition)
+      editor.on("changeSelection", sh.hideSelectionPopover)
+      editor.getSession().on("changeScrollTop", sh.hideSelectionPopover)
+      editor.getSession().on("changeScrollLeft", sh.hideSelectionPopover)
+
+    createPopover: (editor) ->
+      if not contextHandler? or not contextHandler.contextPreviewExists
+        if not katex?
+          initKaTeX(sh.renderSelectionUnderCursor)
+          return
+        sh.renderSelectionUnderCursor()
+  }
 
+  setupPreviewer = (editor, popoverHandler, katexLoader, I18N) ->
+    myKatexLoader = katexLoader
     equationRangeHandler = new EquationRangeHandler(editor)
-
     contextHandler = new ContextHandler(editor, popoverHandler, equationRangeHandler, I18N)
-    exports.reset = -> if contextHandler.contextPreviewExists then contextHandler.destroyEverything()
-
-    sh = selectionHandler = {
-      hideSelectionPopover: ->
-        popoverHandler.destroy()
-        editor.off("changeSelection", sh.hideSelectionPopover)
-        editor.getSession().off("changeScrollTop", sh.hideSelectionPopover)
-        editor.getSession().off("changeScrollLeft", sh.hideSelectionPopover)
-
-      renderSelectionUnderCursor: ->
-        cursorPos = editor.getCursorPosition()
-        cursorPosition = editor.renderer.textToScreenCoordinates(cursorPos.row, cursorPos.column)
-        popoverPosition = {
-          top: "#{cursorPosition.pageY + 24}px"
-          left: "#{cursorPosition.pageX}px"
-        }
-        content = katex.renderToString(
-          editor.getSelectedText(),
-          KATEX_OPTIONS
-        )
-        popoverHandler.show("Preview", content, popoverPosition)
-        editor.on("changeSelection", sh.hideSelectionPopover)
-        editor.getSession().on("changeScrollTop", sh.hideSelectionPopover)
-        editor.getSession().on("changeScrollLeft", sh.hideSelectionPopover)
-
-      createPopover: (editor) ->
-        unless contextHandler.contextPreviewExists
-          unless katex?
-            initKaTeX(sh.renderSelectionUnderCursor)
-            return
-          sh.renderSelectionUnderCursor()
-    }
-    exports.SelectionHandler = selectionHandler
-
     editor.on("changeSelection", contextHandler.handleCurrentContext)
 
+  exports.ContextHandler = ContextHandler
```

Comment:
Do we use all these exports? If some of them are needed for testing purposes only then wrap them into testExport pls:
```
exports.testExport = {
  ContextHandler: ContextHandler
}
```